packages/itwin/grouping-mapping-widget/src/widget/components/Mappings.tsx

```diff
@@ -0,0 +1,115 @@
+  onClickMappingModify?: (mapping: Mapping) => void;
+  displayStrings?: Partial<typeof mappingUIDefaultDisplayStrings>;
+}
+
+const fetchMappings = async (
+  setMappings: React.Dispatch<React.SetStateAction<Mapping[]>>,
+  iModelId: string,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  getAccessToken: GetAccessTokenFn,
+  mappingsClient: IMappingsClient
+) => {
+  try {
+    setIsLoading(true);
+    const accessToken = await getAccessToken();
+    const mappings = await mappingsClient.getMappings(accessToken, iModelId);
+    setMappings(mappings.sort((a, b) => a.mappingName.localeCompare(b.mappingName)));
+  } catch (error: any) {
+    handleError(error.status);
+  } finally {
+    setIsLoading(false);
+  }
+};
+
+export interface MappingsOperationsProps extends GroupingMappingApiConfig {
+  mappingClient: IMappingsClient;
+}
+
+export const useMappingsOperations = ({ iModelId, getAccessToken, mappingClient }: MappingsOperationsProps) => {
+  const [showDeleteModal, setShowDeleteModal] = useState<Mapping | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [mappings, setMappings] = useState<Mapping[]>([]);
+  const [isTogglingExtraction, setIsTogglingExtraction] = useState<boolean>(false);
+
+  useEffect(() => {
+    void fetchMappings(setMappings, iModelId, setIsLoading, getAccessToken, mappingClient);
+  }, [getAccessToken, mappingClient, iModelId]);
+
+  const refresh = useCallback(async () => {
+    setMappings([]);
+    await fetchMappings(setMappings, iModelId, setIsLoading, getAccessToken, mappingClient);
+  }, [getAccessToken, mappingClient, iModelId]);
+
+  const toggleExtraction = useCallback(async (mapping: Mapping) => {
+    try {
+      setIsTogglingExtraction(true);
+      const newState = !mapping.extractionEnabled;
+      const accessToken = await getAccessToken();
+      await mappingClient.updateMapping(accessToken, iModelId, mapping.id, {
+        extractionEnabled: newState,
+      });
+    } catch (error: any) {
+      handleError(error.status);
+    } finally {
+      setIsTogglingExtraction(false);
+    }
+  }, [getAccessToken, iModelId, mappingClient]);
+
+  const onDelete = async (mapping: Mapping) => {
+    const accessToken = await getAccessToken();
+    await mappingClient.deleteMapping(accessToken, iModelId, mapping.id);
+    await refresh();
+  };
+
+  return { mappings, isLoading, refresh, toggleExtraction, onDelete, setShowDeleteModal, showDeleteModal, isTogglingExtraction };
+};
+
+export const Mappings = (props: MappingsProps) => {
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const mappingClient = useMappingClient();
+  const {
+    mappings,
+    isLoading,
+    refresh,
+    toggleExtraction,
+    onDelete,
+    setShowDeleteModal,
+    showDeleteModal,
+    isTogglingExtraction,
+  } = useMappingsOperations({ ...groupingMappingApiConfig, mappingClient });
+
+  return (
```

Comment:
A toast message is displayed if there is an error retrieving mappings. Does the MappingsUI component reflect an error?

Reply:
There is no errored state. I'll try to think of something. Maybe pass in a prop with a error severity or some form of message?

Reply:
Can you simulate an error and paste the Mapping component here?